src/landscape.cpp

```diff
@@ -784,50 +784,60 @@ std::tuple<CommandCost, Money> CmdClearArea(DoCommandFlags flags, TileIndex tile
 	 * shift register (LFSR). This allows a deterministic pseudorandom ordering, but
 	 * still with minimal state and fast iteration. */
 
-	/* Maximal length LFSR feedback terms, from 12-bit (for 64x64 maps) to 24-bit (for 4096x4096 maps).
-	 * Extracted from http://www.ece.cmu.edu/~koopman/lfsr/ */
-	static const uint32_t feedbacks[] = {
-		0xD8F, 0x1296, 0x2496, 0x4357, 0x8679, 0x1030E, 0x206CD, 0x403FE, 0x807B8, 0x1004B2, 0x2006A8, 0x4004B2, 0x800B87
-	};
-	static_assert(lengthof(feedbacks) == 2 * MAX_MAP_SIZE_BITS - 2 * MIN_MAP_SIZE_BITS + 1);
-	const uint32_t feedback = feedbacks[Map::LogX() + Map::LogY() - 2 * MIN_MAP_SIZE_BITS];
+	/* Initialize starting value for LFSR from tick counter to iterate all possible patterns. */
+	static int16_t lfsr_current = (TimerGameTick::counter >> TILE_UPDATE_FREQUENCY_LOG) % (TILE_UPDATE_FREQUENCY - 1) + 1;
+	const uint32_t LFSR_FEEDBACK = 0x8E;  // 8-bit feedback term extracted from http://www.ece.cmu.edu/~koopman/lfsr/
+	static_assert(TILE_UPDATE_FREQUENCY > LFSR_FEEDBACK, "Use appropriate feedback term for the pattern size.");
+	static_assert(2 * LFSR_FEEDBACK > TILE_UPDATE_FREQUENCY, "Use appropriate feedback term for the pattern size.");
+
+	_tile_loop_pattern[0] = TileIndex{0};
+	for (uint i = 1; i < TILE_UPDATE_FREQUENCY; i++) {
+		lfsr_current = (lfsr_current >> 1) ^ (-(lfsr_current & 1) & LFSR_FEEDBACK);
 
-	/* We update every tile every TILE_UPDATE_FREQUENCY ticks, so divide the map size by 2^TILE_UPDATE_FREQUENCY_LOG = TILE_UPDATE_FREQUENCY */
-	static_assert(2 * MIN_MAP_SIZE_BITS >= TILE_UPDATE_FREQUENCY_LOG);
-	uint count = 1 << (Map::LogX() + Map::LogY() - TILE_UPDATE_FREQUENCY_LOG);
+		/* lfsr_current is a location in the pattern, convert it to map coordinates. */
+		_tile_loop_pattern[i] = TileXY(lfsr_current % TILE_LOOP_PATTERN_SIZE, lfsr_current / TILE_LOOP_PATTERN_SIZE);
+	}
+}
 
-	TileIndex tile = _cur_tileloop_tile;
-	/* The LFSR cannot have a zeroed state. */
-	assert(tile != 0);
+void RunTileLoop()
+{
+	PerformanceAccumulator framerate(PFE_GL_LANDSCAPE);
 
-	/* Manually update tile 0 every TILE_UPDATE_FREQUENCY ticks - the LFSR never iterates over it itself.  */
+	/* Every TILE_UPDATE_FREQUENCY ticks generate a new tile pattern just to make it a bit more random. */
 	if (TimerGameTick::counter % TILE_UPDATE_FREQUENCY == 0) {
-		_tile_type_procs[GetTileType(0)]->tile_loop_proc(TileIndex{});
-		count--;
+		GenerateTileLoopPattern();
 	}
 
-	while (count--) {
-		_tile_type_procs[GetTileType(tile)]->tile_loop_proc(tile);
+	/* Having sizes equally divisible helps to avoid boundary checks in the loop. */
+	assert(Map::SizeX() % TILE_LOOP_PATTERN_SIZE == 0);
+	assert(Map::SizeY() % TILE_LOOP_PATTERN_SIZE == 0);
 
-		/* Get the next tile in sequence using a Galois LFSR. */
-		tile = TileIndex{(tile.base() >> 1) ^ (-(int32_t)(tile.base() & 1) & feedback)};
-	}
+	auto pattern_tile = _tile_loop_pattern[TimerGameTick::counter % TILE_UPDATE_FREQUENCY];
```

Comment:
You could try adding an offset based on the region index. That way every region will use a different pattern and hawk-eyed users among us should not even recognize the patterns if there's enough variety.

You might want to make the size of _tile_loop_pattern a prime number to avoid patterns lining up exactly.